sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureCliCredential.java

```diff
@@ -83,7 +83,8 @@ public Mono<AccessToken> getToken(TokenRequestContext request) {
         return identityClient.authenticateWithAzureCli(request)
             .doOnNext(token -> LoggingUtil.logTokenSuccess(LOGGER, request))
             .doOnError(error -> LoggingUtil.logTokenError(LOGGER, identityClient.getIdentityClientOptions(), request,
-                error));
+                error))
+            .onErrorMap(error -> new CredentialUnavailableException("Azure CLI authentication is not available", error));
```

Comment:
This and the sync try/catch are potentially mapping different exception types. `onErrorMap` defaults to mapping all `Throwable`s, we should look into using the overload method `onErrorMap(Class<? extends Throwable>, Function)`.

Reply:
good catch!